src/split_loops.jl

```diff
@@ -74,12 +74,12 @@ function lower_and_split_loops(ls::LoopSet)
             ls_2_lowered = if length(remaining_ops) > 1
                 lower_and_split_loops(ls_2)
             else
-                lower(ls_2, unrolled_2, tiled_2, vectorized_2, U_2, T_2)
+                lower(ls_2, order_2, unrolled_2, tiled_2, vectorized_2, U_2, T_2)
```

Comment:
I think this and the next must have been a bug? If so, it's not caught by current tests. I am unsure how to go about designing one.

Reply:
Yes, you're right, thanks for catching that.
Here is a testcase:
```julia
function twogemms!(Ab, Bb, Cb, A, B)
    M, N = size(C); K = size(B,1)
    @avx for m in 1:M, k in 1:K, n in 1:N
        Ab[m,k] += Cb[m,n] * B[k,n]
        Bb[k,n] += A[m,k] * Cb[m,n]
    end
end
```
This predictably throws a method error:
```julia
julia> M = 100;

julia> A = rand(M,M); B = rand(M,M); Cb = rand(M,M); Ab = similar(A); Bb = similar(B);

julia> twogemms!(Ab, Bb, Cb, A, B)
ERROR: MethodError: no method matching lower(::LoopVectorization.LoopSet, ::Symbol, ::Symbol, ::Symbol, ::Int64, ::Int64)
```
The motivation was that the autodiff code could produce loops like this. This particular example is what it should be producing as the reverse diff pass for `C = A * B` (`Cb`, `Ab`, and `Bb` being the partials with respect to `C`, `A`, and `B`).
It'd be more efficient to split it.
I should've added tests, also to make sure it really does split.

I shouldn't have merged until I actually added tests for the new functionality, even if all existing tests passed.

Reply:
I implemented your fixes and added `twogemms!` to the unit tests.
I should probably add a `threegemms!` as well to hit the last remaining un-evaluated line in `split_loops!`. https://github.com/chriselrod/LoopVectorization.jl/blob/0e0c418f4fa9a976c09a0fb2b9cd7505632c3e16/src/split_loops.jl#L75
Thanks again!